hw/ip/keymgr/doc/_index.md

```diff
@@ -0,0 +1,211 @@
+If a command is valid during the current working state, it is processed and acknowledged when complete.
+If a command is invalid, the key manager FSM processes with random, dummy data, but does not update working state or relevant output registers.
+
+For each valid command, a set of inputs are selected and sequenced to the KMAC module.
+
+## Generating Output Key
+The generate output command is composed of 2 options
+*  Generate output key for software
+*  Generate output key for hardware
+
+The hardware command is meant specifically for symmetric side load use cases.
+When this command is issued, the output of the KMAC invocation is not stored in software visible registers, but instead in hardware registers that directly output to symmetric primitives such as AES, HMAC and KMAC.
+
+## KMAC Operations
+All invoked KMAC operations expect the key in two-shares.
+This means the working states, even though functionally 256b, are maintained as 512b.
+
+For advance-state and generate-output commands, the KMAC emitted output are also in 2-shares.
+It is up to the software and the symmetric primitives to decide if the key can be processed as shares or XOR’d to reveal the actual value.
+
+## Block Diagram
+The following is a high level block diagram of the key manager.
+
+![Key Manager Block Diagram](keymgr_block_diagram.svg)
+
+## Design Details
+
+Key manager is primarily composed of two components:
+*  keymgr_ctrl
+*  keymgr_kmac_if
+
+### Key Manager Control
+
+The key manager control block manages the working state, sideload key  updates, as well as what commands are valid in each state.
+It also handles the life cycle keymgr_en input, which helps disable the entire key manager function in the event of an escalation.
+
+![Key Manager Control Block Diagram](keymgr_control_diagram.svg)
+
+
+### KMAC Interface Control
+
+The KMAC interface control represents the bulk of key manager logic.
+Based on input from key manager control, this module selects the inputs for each given command and sequences the data to KMAC.
+For the sake of simplicity, the diagram below shows KMAC internal, however it can also be an external module.
+
+![Key Manager KMAC Interface Block Diagram](keymgr_kmac_if_diagram.svg)
+
+## Hardware Interfaces
+{{< hwcfg "hw/ip/keymgr/data/keymgr.hjson" >}}
+
+# Programmers Guide
+
+## Initialize
+
+## Advance or Generate
+Software selects a command and triggers a “start” to.
+If the command is valid and successful, key manager indicates done and no errors.
+If the command is invalid or unsuccessful, key manager indicates done with error.
+Regardless of the validity of the command, the hardware sequences are triggered to avoid leaking timing information.
+
+The software is able to read what state the key manager is currently in, however it never has access to the associated secret value.
+
+When selecting the generate-output-hw command, software must also indicate which peripheral the output is targeted.
```

Comment:
suggest: 
"When issuing the generate-output-hw command, software must select a destination peripheral."  (tbd, peripheral vs primitive here?)
"At the conclusion of the command, the appropriate set of valid signals are forwarded to the selected destination primitive."
"The key and valid signals remain asserted to the selected destination until software explicitly disables the output via XYZ command, or issues another generate-output-hw command with a different destination primitive." (or peripheral, maybe?)

Reply:
thanks! yes this is better.